packages/flutter_tools/lib/src/base/analyze_size.dart

```diff
@@ -116,16 +116,20 @@ class SizeAnalyzer {
     final _SymbolNode apkAnalysisRoot = _parseUnzipFile(zipFile);
 
     // Convert an AOT snapshot file into a map.
-    final Map<String, dynamic> processedAotSnapshotJson = treemapFromJson(
-      json.decode(aotSnapshot.readAsStringSync()),
-    );
-    final _SymbolNode aotSnapshotJsonRoot = _parseAotSnapshot(processedAotSnapshotJson);
-    for (final _SymbolNode firstLevelPath in apkAnalysisRoot.children) {
-      _printLibChildrenPaths(firstLevelPath, '', aotSnapshotJsonRoot, _kZipSizeMaxDepth, 0);
+    final Object? decodedAotSnapshot = json.decode(aotSnapshot.readAsStringSync());
+    if (decodedAotSnapshot == null) {
+      throwToolExit('AOT snapshot is invalid for analysis');
+    }
+    final Map<String, Object?> processedAotSnapshotJson = treemapFromJson(decodedAotSnapshot);
+    final _SymbolNode? aotSnapshotJsonRoot = _parseAotSnapshot(processedAotSnapshotJson);
+    if (aotSnapshotJsonRoot != null) {
```

Comment:
I mostly ignored null nodes or missing keys instead of throwing, I can change it if we want to be more aggressive with the expected formatting instead of silently handling it, if preferred.